### DISCUSSION TITLE
Effect of batch size and rmin on MTP training and MD stability

### DISCUSSION THREAD

**Discussion by lhycms** [lhycms/AI2Pot#21] [Show and tell]
# 1. Effect of `batch_size`：Lower batch size, higher accuracy, but loss of atoms

When training an `MTP` model for Ge-Sb-Te with ZBL enabled, I found that, under the same training setting of 200 epochs, `batch_size=1` gives lower errors than `batch_size=16`.

The corresponding energy, force, and virial parity plots are shown below.

`batch_size = 1`:

<img width="2006" height="561" alt="image" src="https://github.com/user-attachments/assets/c2b6b4bd-9097-4ed0-b955-8788047a72da" />

`batch_size = 16`:

<img width="1581" height="447" alt="image" src="https://github.com/user-attachments/assets/c9fabe42-4acf-4131-b5cd-ff640dc313d9" />

However, although `batch_size=1` gives better fitting accuracy, atom loss occurs during MD simulations. In contrast, the model trained with `batch_size=16` does not show this issue under the same MD conditions.

This **"lose atoms"** behavior and possible solutions have also been discussed in #19.

# 2. Effect of `rmin`: Only ZBL inner `rmin`

I also tested the effect of the `rmin` parameter in the `MTP` model and found that `rmin` has a noticeable influence on the fitting accuracy.

In the current implementation, the MTP contribution is excluded for the short-range region with

$$
r < r_{\min},
$$

and this region is handled entirely by the ZBL potential. Therefore, changing `rmin` effectively changes the range over which the MTP and ZBL contributions are applied.

The corresponding energy, force, and virial parity plots are shown below.

`batch_size = 16`, `rmin = 1.0`:

<img width="1582" height="446" alt="image" src="https://github.com/user-attachments/assets/a94846d8-e9dc-4789-8425-993bcb796dfc" />

`batch_size = 16`, `rmin = 1.5`:

<img width="1580" height="449" alt="image" src="https://github.com/user-attachments/assets/602ac957-73ad-41d6-ab0b-285d0f30b252" />

`batch_size = 1`, `rmin = 1.5`:

Results to be tested.

# 3. MD stability tests

The next step is to compare the MD stability of models trained with:

* `rmin = 1.0`
* `rmin = 1.5`

and check whether atom loss occurs under the same MD conditions.

These tests will help determine whether increasing `rmin` can improve short-range stability while maintaining acceptable fitting accuracy.
